instrumentation/spring/spring-boot-resources/testing/src/test/java/io/opentelemetry/instrumentation/spring/resources/TestBootInfClassesResource.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.spring.resources;
+
+import static io.opentelemetry.semconv.resource.attributes.ResourceAttributes.SERVICE_NAME;
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigProperties;
+import io.opentelemetry.sdk.resources.Resource;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class TestBootInfClassesResource {
+  @Mock ConfigProperties config;
+
+  @Test
+  void testServiceName() {
+    // verify that the test app, that is added as a dependency to this project, has the expected
+    // layout
+    assertThat(getClass().getResource("/application.properties")).isNull();
+    assertThat(getClass().getResource("/BOOT-INF/classes/application.properties")).isNotNull();
+
+    SpringBootServiceNameDetector guesser = new SpringBootServiceNameDetector();
+    Resource result = guesser.createResource(config);
```

Comment:
```suggestion
    SpringBootServiceNameDetector detector = new SpringBootServiceNameDetector();
    Resource result = detector.createResource(config);
```

Reply:
This is copy pasted from
https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/195fd38c2c5eafcf918f839ee51f68bc6f4b3d03/instrumentation/spring/spring-boot-resources/library/src/test/java/io/opentelemetry/instrumentation/spring/resources/SpringBootServiceNameDetectorTest.java#L49-L50
WDYT about renaming this in both tests after this is merged?